procbuild/pr_list/__init__.py

```diff
@@ -26,37 +33,45 @@ def fetch_PRs(user, repo, state='open'):
     config.update(fields)
 
     data = []
-    page_data = True
+    page_data = []
 
-    url = 'https://api.github.com/repos/{user:s}/{repo:s}/pulls'.format(**config)
+    base_url = 'https://api.github.com/repos/''
+    url = f'{base_url}{config["user"]}/{config["repo"]}/pulls'
     http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED')
 
-    while page_data:
-        fetch_status = 'Fetching page {page:d} (state={state:s})'.format(**fields) + \
-                       ' from {user:s}/{repo:s}...'.format(**config)
+    while len(data) == 0 or len(page_data) == responses_per_page:
+        fetch_status = ('Fetching page {page:d} (state={state:s})'
+                        ' from {user:s}/{repo:s}...').format(**config)
         print(fetch_status)
 
         response = http.request('GET', url, fields=fields,
                                 headers={'user-agent': 'scipy-procbuild/0.1'})
 
         fields['page'] += 1
+        config.update(fields)
 
         page_data = json.loads(response.data.decode('utf-8'))
 
         if 'message' in page_data and page_data['message'] == "Not Found":
-            page_data = []
-            print('Warning: Repo not found ({user:s}/{repo:s})'.format(**config))
+            print(('Warning: Repo not found '
+                   '({user:s}/{repo:s})').format(**config))
+            break
+        elif len(page_data) == 0:
```

Comment:
This reads a bit strangely as an `elif`: it can be a standalone if, which makes its intention more clear.  Perhaps also move earlier, right after the while?

Reply:
I don't really understand what your reasoning is. I can make it the if instead of the elif, as it's a simpler condition, but it cannot be run until we actually get page_data which happens immediately before this set of conditionals. 

I'll add a comment to make it clearer why these should be tied together.

Reply:
Note: right after the while would mean we never entered the request, because the page_data needs to be initially of length 0 in order for the condition on the while to be valid.